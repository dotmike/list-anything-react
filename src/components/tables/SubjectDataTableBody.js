import React, { Component } from "react";
import PropTypes from "prop-types";
import { Table, Loader } from "semantic-ui-react";

// TODO: deixar mais genÃ©rico
class SubjectDataTableBody extends Component {
  shouldComponentUpdate = nextProps => {
    const { loading, data, fields } = this.props;
    return (
      loading !== nextProps.loading ||
      data !== nextProps.data ||
      fields !== nextProps.fields
    );
  };

  render() {
    const { loading, data, fields, onTableRowClick, renderCell } = this.props;
    return (
      <Table.Body style={{ cursor: "pointer" }}>
        {loading && (
          <Table.Row>
            <Table.Cell width={16}>
              <Loader active inline="centered" />
            </Table.Cell>
          </Table.Row>
        )}
        {!loading &&
          !data.length && (
            <Table.Row>
              <Table.Cell width={16}>Nothing was found.</Table.Cell>
            </Table.Row>
          )}
        {!loading &&
          data.map(sd => (
            <Table.Row
              key={sd._id}
              to={`/subject-data/${sd._id}`}
              onClick={onTableRowClick}
            >
              {fields.map(field => field.show_in_list && renderCell(sd, field))}
            </Table.Row>
          ))}
      </Table.Body>
    );
  }
}

SubjectDataTableBody.propTypes = {
  loading: PropTypes.bool.isRequired,
  data: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string
    })
  ).isRequired,
  fields: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string
    })
  ).isRequired,
  onTableRowClick: PropTypes.func.isRequired,
  renderCell: PropTypes.func.isRequired
};

export default SubjectDataTableBody;
