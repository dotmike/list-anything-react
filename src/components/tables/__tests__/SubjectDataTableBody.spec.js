import React from "react";
import { shallow } from "enzyme";

import SubjectDataTableBody from "../SubjectDataTableBody";

const defaultProps = {
  loading: false,
  data: [],
  fields: [],
  onTableRowClick: () => {},
  renderCell: () => {}
};

describe("SubjectDataTableBody", () => {
  it("renders correctly", () => {
    defaultProps.results = [];

    const wrapper = shallow(<SubjectDataTableBody {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
