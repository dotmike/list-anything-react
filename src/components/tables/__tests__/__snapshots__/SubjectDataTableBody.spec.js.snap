// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectDataTableBody renders correctly 1`] = `
<TableBody
  as="tbody"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <TableRow
    as="tr"
    cellAs="td"
  >
    <TableCell
      as="td"
      width={16}
    >
      Nothing was found.
    </TableCell>
  </TableRow>
</TableBody>
`;
